Add support for using full width containers on component preview pages

MartinJJones opened the pull request:
## What
- Add support for using full width containers on component preview pages
- Update the components below to use full width layout on the preview page
  - layout_super_navigation_header
  - layout_header
  - layout_footer
  - service_navigation
  - cross-service-header
- Update the preview page spacing to ensure the components accurately reflect how they will appear when used in application without any other stying applied

## Why
The main motivation for this change is to fix an issue on the [super navigation component preview page](https://components.publishing.service.gov.uk/component-guide/layout_super_navigation_header/preview) where the logo and menu button appear to overlap as shown in the screenshot below:

<img width="483" height="166" alt="super-nav-overlap" src="https://github.com/user-attachments/assets/2dc4e222-d9d2-445a-ad2f-f810884b8476" />

This can cause unexpected issues in Percy as it is set to take mobile screenshots at a width of `375px` by default - https://www.browserstack.com/docs/percy/references/config-options#all-configuration-options

In the case of the super navigation menu, it has styling in place so that when the available screen width is `360px` or below, the [menu buttons will be smaller](https://github.com/alphagov/govuk_publishing_components/blob/main/app/assets/stylesheets/govuk_publishing_components/components/_layout-super-navigation-header.scss#L206-L209).

At 375px this media query is not applied, but the available width is only `325px` (375px - (20px padding + 30px margin))

Allowing the super navigation menu to take up with full width of the page fixes this issue and ensures the screenshots captured in Percy are accurate, it is also consistent with the approach used in other component guide preview pages.

### Examples where full width previews are used 

#### One Login

- https://govuk-one-login.github.io/service-header/dist/preview-rebranded.html

#### Design System

- https://govuk-frontend-review.herokuapp.com/components/header/preview
- https://govuk-frontend-review.herokuapp.com/components/footer/preview
- https://govuk-frontend-review.herokuapp.com/components/service-navigation/preview

## Visual Changes

Percy is showing visual changes for 89/90 components, the exception being `/public`, this is expected as the layout and styling changes impacts all component preview pages.

AshGDS reviewed (commented):
Looks good to me 👍  A couple of the full width examples have `This is difficult to preview because the preview windows are constrained` in their descriptions, so it's probably worth removing that now?

andysellick commented on app/views/layouts/govuk_publishing_components/application.html.erb:
```diff
@@ -1,19 +1,24 @@
+    <% 
+      govuk_container_class = @component_doc.capture_full_width_screenshot? ? "govuk-full-width-container" : "govuk-width-container"
+    %>
+    <%= tag.div id: "wrapper", class: govuk_container_class do %>
```
For components that use the new full width, would it be worth putting a little bit of spacing between the edge of the screen and the preview name?

<img width="585" height="159" alt="Screenshot 2025-08-04 at 08 51 09" src="https://github.com/user-attachments/assets/27990e0e-4706-4ee9-a533-a278f2c55291" />

MartinJJones replied:
Thanks for the review, I've fixed the typo on the 2nd commit description and also increased the spacing between the component title and edge of the screen when the preview page uses the full width container.

andysellick reviewed (commented):
Looks good, nice improvement. Still going through the visual diffs to check, but everything seems good so far.

Nitpick: typo in the 2nd commit description? `..how the component will appear on the page without and custom styles added.`

andysellick commented:
@MartinJJones surprised the visual diff isn't showing any changes for the mobile screenshots - would you expect that?

AshGDS commented:
Thanks, the tests look good 👍 

I just merged a change to `service_navigation` which has the same "This is difficult to preview because the preview windows are constrained" message which will need to be removed as part of this PR.

MartinJJones commented:
> I just merged a change to service_navigation which has the same "This is difficult to preview because the preview windows are constrained" message which will need to be removed as part of this PR.

Thanks @AshGDS, I've rebased and removed the message above from the `service_navigation` component

MartinJJones commented on app/views/layouts/govuk_publishing_components/application.html.erb:
```diff
@@ -1,19 +1,24 @@
-      <%= yield %>
-    </main>
+    <% 
+      govuk_container_class = @component_doc.capture_full_width_screenshot? ? "govuk-full-width-container" : "govuk-width-container"
```
Thanks, yeah I think this is much cleaner, I've moved to `compontent_doc.rb` as suggested
